### DISCUSSION TITLE
how to prevent Hot Module Reloading

### DISCUSSION THREAD

**Discussion by amit0shakya** [vercel/next.js#12539] [Help] [answered]
I have an Interactive web application, where user can upload face and to fun stuff but due to HMR reloading it reloads stylesheet within 30 seconds and whole user data losted. I am trying pretty hard to do this but not getting way to fix this

**Comment by Timer** [comment]
Could you elaborate more? Is the page reloading when it shouldn't? Did you edit styles?

I'd like to understand the problem more to provide advice.

**Reply by amit0shakya** [reply]
I make request on window resize, but it is reloading due to HMR as well. It is realoading automatically with in 30 seconds this project is on server http://www.only4laugh.com/create/jokes/world_record

**Comment by LukasBombach** [comment]
HMR is part of the webpack dev server. You should be able to disable it in your webpack config:

`next.config.json`

```js
module.exports = {
  webpack: (config) => {
    config.devServer.hot = false;
    return config
  },
}
```

https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config

But I am wondering if this is really what you want. Can you elaborate your problem a little more?

**Reply by amit0shakya** [reply]
My web based project on server refreshing continuously due to HMR reloading, That's why I want to prevent it http://www.only4laugh.com/create/jokes/world_record, this page is also reload on window resize.

**Reply by elsigh** [reply]
@LukasBombach 

This didn't work for me - I get the following error when starting the dev server:
```TypeError: Cannot set property 'hot' of undefined```

I want to disable hmr because it seems that every time I save a file (multiple times while I noodle around writing code) I come back to my app (which loads things on the network on load) and it loads one time for each time I'd clicked save in my editor.

Also when I create some errors by using bad var names or whatnot, I'll find that my browser goes into a death loop, reloading with the error again and again (and sending network requests each time). Sometimes I end up on the error page but I cannot figure out the logic for that.

I feel like I must be doing something wrong somwhere.

I'll say that when it works well, it's sweet, but the pain of having to use the Chrome task manager to kill my tab and then refresh is just an uncomfortable dev experience. Ideas? Thanks!

**Reply by Timer** [reply]
Disabling HMR is _never_ what someone wants. As such, I'm collapsing this suggestion. @elsigh please try the new React Refresh support on canary which makes your app reloading much better.

**Comment by rafaelalmeidatk** [comment, marked as answer]
Looks like you are running your server in dev mode (`next dev`). For production you should build the project (`next build`) and start the server with `next start`.
